File: judge/tests/malicious/read_passwd.c
// Malicious program attempting to read /etc/passwd
#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE *f = fopen("/etc/passwd", "r");
    if (f == NULL) {
        printf("BLOCKED: Cannot open /etc/passwd\n");
        return 1;
    }

    char buffer[1024];
    if (fgets(buffer, sizeof(buffer), f) != NULL) {
        printf("SECURITY BREACH: Read /etc/passwd: %s\n", buffer);
        fclose(f);
        return 0;
    }

    fclose(f);
    return 1;
}

File: judge/tests/malicious/read_passwd.rs
// Malicious Rust program attempting to read /etc/passwd
use std::fs;

fn main() {
    match fs::read_to_string("/etc/passwd") {
        Ok(contents) => {
            println!("SECURITY BREACH: Read /etc/passwd");
            println!("First line: {}", contents.lines().next().unwrap_or(""));
            std::process::exit(0);
        }
        Err(e) => {
            println!("BLOCKED: Cannot read /etc/passwd: {}", e);
            std::process::exit(1);
        }
    }
}
